tests/misc/test_readline.py

```diff
@@ -21,144 +21,256 @@
-
-    def test_rl_unix_word_rubout(self, mocked_qapp):
-        """Set a selected text, delete it, see if it comes back with yank."""
-        mocked_qapp.focusWidget = mock.Mock(return_value=self.qle)
-        self._set_selected_text("delete test")
-        self.bridge.rl_unix_word_rubout()
-        self.qle.cursorWordBackward.assert_called_with(True)
-        assert self.bridge._deleted[self.qle] == "delete test"
-        self.qle.del_.assert_called_with()
-        self.bridge.rl_yank()
-        self.qle.insert.assert_called_with("delete test")
-
-    def test_rl_kill_word(self, mocked_qapp):
-        """Set a selected text, delete it, see if it comes back with yank."""
-        mocked_qapp.focusWidget = mock.Mock(return_value=self.qle)
-        self._set_selected_text("delete test")
-        self.bridge.rl_kill_word()
-        self.qle.cursorWordForward.assert_called_with(True)
-        assert self.bridge._deleted[self.qle] == "delete test"
-        self.qle.del_.assert_called_with()
-        self.bridge.rl_yank()
-        self.qle.insert.assert_called_with("delete test")
-
-    def test_rl_yank_no_text(self, mocked_qapp):
-        """Test yank without having deleted anything."""
-        mocked_qapp.focusWidget = mock.Mock(return_value=self.qle)
-        self.bridge.rl_yank()
-        assert not self.qle.insert.called
+def bridge():
+    """Fixture providing a ReadlineBridge."""
+    return readline.ReadlineBridge()
+
+
+def test_none(bridge, qtbot):
+    """Call each rl_* method with a None focusWidget."""
+    assert QApplication.instance().focusWidget() is None
+    for name, method in inspect.getmembers(bridge, inspect.ismethod):
+        if name.startswith('rl_'):
+            method()
+
+
+@pytest.mark.parametrize('text, expected', [('f<oo>bar', 'fo|obar'),
+                                            ('|foobar', '|foobar')])
+def test_rl_backward_char(text, expected, lineedit, bridge):
+    """Test rl_backward_char."""
+    lineedit.set_aug_text(text)
+    bridge.rl_backward_char()
+    assert lineedit.aug_text() == expected
+
+
+@pytest.mark.parametrize('text, expected', [('f<oo>bar', 'foob|ar'),
+                                            ('foobar|', 'foobar|')])
+def test_rl_forward_char(text, expected, lineedit, bridge):
+    """Test rl_forward_char."""
+    lineedit.set_aug_text(text)
+    bridge.rl_forward_char()
+    assert lineedit.aug_text() == expected
+
+
+@pytest.mark.parametrize('text, expected', [('one <tw>o', 'one |two'),
+                                            ('<one >two', '|one two'),
+                                            ('|one two', '|one two')])
+def test_rl_backward_word(text, expected, lineedit, bridge):
+    """Test rl_backward_word."""
+    lineedit.set_aug_text(text)
+    bridge.rl_backward_word()
+    assert lineedit.aug_text() == expected
+
+
+@pytest.mark.parametrize('text, expected', [
+    fixme(('<o>ne two', 'one| two')),
+    ('<o>ne two', 'one |two'),  # wrong
```

Comment:
Should those params marked with a `#wrong` comment also be decorated with `fixme`?